packages/utils/src/snark-artifacts/snark-artifacts.node.ts

```diff
@@ -47,84 +47,24 @@ async function maybeDownload(url: string) {
-function MaybeGetSnarkArtifacts(
-    proof: Proof.POSEIDON,
-    version?: Version
-): (numberOfInputs: number) => Promise<SnarkArtifacts>
-function MaybeGetSnarkArtifacts(proof: Proof.SEMAPHORE): (treeDepth: number) => Promise<SnarkArtifacts>
-function MaybeGetSnarkArtifacts(proof: Proof, version?: Version) {
-    switch (proof) {
-        case Proof.POSEIDON:
-            return async (numberOfInputs: number) =>
-                GetSnarkArtifactUrls({ proof, numberOfInputs, version }).then(maybeGetSnarkArtifacts)
-        case Proof.SEMAPHORE:
-            return async (treeDepth: number) =>
-                GetSnarkArtifactUrls({ proof, treeDepth, version }).then(maybeGetSnarkArtifacts)
-
-        case Proof.EDDSA:
-            return async () => GetSnarkArtifactUrls({ proof, version }).then(maybeGetSnarkArtifacts)
-
-        default:
-            throw new Error("Unknown proof type")
-    }
-}
-
-/**
- * Downloads {@link @zk-kit/eddsa-proof!generate | EdDSA} snark artifacts (`wasm` and `zkey`) files if not already present in OS tmp folder.
- * @example
- * ```
- * {
- *   wasm: "/tmm/@zk-kit/eddsa-artifacts@latest/eddsa.wasm",
- *   zkey: "/tmp/@zk-kit/eddsa-artifacts@latest/eddsa.zkey"
- * }
- * ```
- * @returns {@link SnarkArtifacts}
- */
-export const maybeGetEdDSASnarkArtifacts = MaybeGetSnarkArtifacts(Proof.EDDSA)
-
 /**
- * Downloads {@link @zk-kit/poseidon-proof!generate | Poseidon} snark artifacts (`wasm` and `zkey`) files if not already present in OS tmp folder.
- * @param numberOfInputs - The number of inputs to hash
+ * Downloads SNARK artifacts (`wasm` and `zkey`) files if not already present in OS tmp folder.
  * @example
  * ```ts
  * {
- *   wasm: "/tmm/@zk-kit/poseidon-artifacts@latest/poseidon-2.wasm",
- *   zkey: "/tmp/@zk-kit/poseidon-artifacts@latest/poseidon-2.zkey"
- * }
- * ```
- * @returns {@link SnarkArtifacts}
- */
-export const maybeGetPoseidonSnarkArtifacts = MaybeGetSnarkArtifacts(Proof.POSEIDON)
-
-/**
- * Downloads {@link https://github.com/semaphore-protocol/semaphore/tree/main/packages/proof | Semaphore} snark artifacts (`wasm` and `zkey`) files if not already present in OS tmp folder.
- * @param treeDepth - The depth of the tree
- * @example
- * ```ts
- * {
- *   wasm: "/tmm/@zk-kit/semaphore-artifacts@latest/semaphore-3.wasm",
+ *   wasm: "/tmp/@zk-kit/semaphore-artifacts@latest/semaphore-3.wasm",
  *   zkey: "/tmp/@zk-kit/semaphore-artifacts@latest/semaphore-3.zkey"
  * }
  * ```
  * @returns {@link SnarkArtifacts}
  */
-export const maybeGetSemaphoreSnarkArtifacts = MaybeGetSnarkArtifacts(Proof.SEMAPHORE)
+export default async function maybeGetSnarkArtifacts(
+    ...pars: Parameters<typeof _maybeGetSnarkArtifacts>
+): Promise<SnarkArtifacts> {
+    const { wasm, zkey } = await _maybeGetSnarkArtifacts(...pars)
+
+    return {
+        wasm: await maybeDownload(wasm),
+        zkey: await maybeDownload(zkey)
```

Comment:
this will fetch artifacts sequentially, instead of concurrently (if both are e.g 3MB, you wait to dl 3MB of wasm before starting 3MB of zkey).
I had observed that Concurrent download of wasm and zkey seemed to give a big download speed benefit
https://github.com/privacy-scaling-explorations/zk-kit/pull/240#discussion_r1557401950